wen-restart/src/last_voted_fork_slots_aggregate.rs

```diff
@@ -217,37 +193,18 @@ impl LastVotedForkSlotsAggregate {
                 }
             }
         }
-        false
     }
 
-    fn update_epoch_info(
-        &mut self,
-        from: &Pubkey,
-        last_vote_epoch: Epoch,
-        old_last_vote_epoch: Option<Epoch>,
-    ) {
-        if Some(last_vote_epoch) < old_last_vote_epoch {
-            // We only have two entries so old epoch must be the second one.
-            let entry = self.epoch_info_vec.last_mut().unwrap();
+    fn update_epoch_info(&mut self, from: &Pubkey, last_vote_epoch: Epoch) {
+        for entry in self.epoch_info_vec.iter_mut() {
             if let Some(stake) = self.root_bank.epoch_node_id_to_stake(entry.epoch, from) {
-                entry.actively_voting_for_this_epoch_stake = entry
-                    .actively_voting_for_this_epoch_stake
-                    .checked_sub(stake)
-                    .unwrap();
-            }
-        } else {
-            for entry in self.epoch_info_vec.iter_mut() {
-                if let Some(stake) = self.root_bank.epoch_node_id_to_stake(entry.epoch, from) {
-                    if old_last_vote_epoch.is_none() {
-                        entry.actively_voting_stake =
-                            entry.actively_voting_stake.checked_add(stake).unwrap();
-                    }
-                    if Some(entry.epoch) > old_last_vote_epoch && entry.epoch <= last_vote_epoch {
-                        entry.actively_voting_for_this_epoch_stake = entry
-                            .actively_voting_for_this_epoch_stake
-                            .checked_add(stake)
-                            .unwrap();
-                    }
+                entry.actively_voting_stake =
+                    entry.actively_voting_stake.checked_add(stake).unwrap();
+                if entry.epoch <= last_vote_epoch {
+                    entry.actively_voting_for_this_epoch_stake = entry
+                        .actively_voting_for_this_epoch_stake
```

Comment:
hmm I thought `actively_voting_for_this_epoch_stake` was only for stake voting exactly for this particular epoch `last_vote_epoch`, not all epochs less than `last_vote_epoch`

This is different than the `actively_voting_stake` for an epoch `N` is the one that tracks all voting validators in epoch N, regardless if they're voting exactly on  epoch `N`

Reply:
It's because we've only given the epoch for last_vote. It is impossible that a validator voted for some slot in epoch N but not any slot in epoch N-1.

If you think about how we use it, we exit when we have >80% stake for any epoch where any slot has >33% stake. If we are checking epoch N because enough people voted for it, surely we need >80% stake for epoch N+1 as well?

Reply:
> If you think about how we use it, we exit when we have >80% stake for any epoch where any slot has >33% stake.

Yeah that makes sense, but just based on the comments for these variables: https://github.com/anza-xyz/agave/blob/e7ee5ad3fe5ef5f0a788ac7400d948663485de0e/wen-restart/src/last_voted_fork_slots_aggregate.rs#L28-L32 the `actively_voting_stake` should be what's used for tracking the 80%, the `actively_voting_for_this_epoch_stake` is what's used for tracking the 33%

Reply:
Updated both comments, does it look better now?